### PR TITLE
Update package.json, added missing export types for module

### PR DESCRIPTION
**Pull request by JimmyNair**
Added missing type output. Resolved vscode error for not finding type declaration for react-d3-tree package

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
   "exports": {
     ".": {
       "import": "./lib/esm/index.js",
-      "require": "./lib/cjs/index.js"
+      "require": "./lib/cjs/index.js",
+      "types": "./lib/types/index.t.ts",
```
Comment (transfer1992) [inline package.json:19]: I believe it should be: `"types": "./lib/types/index.d.ts"`

**Comment by hyper-dot** [comment]
Why it is not merged till today :cry:

File: package.json (hunk 1 above)
Comment (bkrem) [reply package.json:19]: Will correct this before publishing 👍

**Comment by bkrem** [comment]
Hey folks, sorry about this! I've had very little time to maintain this lib for the past year but tried to keep things moving on urgent fixes, this one clearly got away from me. Feel free to nudge as @hyper-dot has here on major annoyances like this.

I was under the impression that the existing root level `types` field was sufficient but clearly there was a gap in my TS dogfooding, presumably when using strictly ES modules.

Thank you for the original fix PR @JimmyNair 🙏 Will ping back here when published.

File: package.json (hunk 1 above)
Comment (transfer1992) [reply package.json:19]: Hey!
I just caught you merging this right now and I am here to correct my mistake. Now I know better.

The `types` should be outside of `exports`, in the main level of `package.json`. It also can (not sure if should) have an absolute path (no `./` at the beginning).

So finally: `"types": "lib/types/index.t.ts",` on the main level.

Sorry for this!

**Comment by bkrem** [comment]
Fixed in https://github.com/bkrem/react-d3-tree/releases/tag/v3.6.2

**Comment by transfer1992** [comment]
Hey!
I just caught you merging this right now and from today's perspective I think, that the `types` property should be placed on the main level of the `package.json` to work properly.

https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package

**Comment by transfer1992** [comment]
Ah, sorry, I can see that you already have the main lavel entry for types. Does this one in `exports` do anything then? Is is actually correct?

**Comment by transfer1992** [comment]
Hey! Subscribers of this PE may have received some emails about me posting some concerns, but I just have educated myself and all is good. Sorry for the mess. Thanks for merging this!

**Pull request merged**